OpenSim/Actuators/Millard2012EquilibriumMuscle.cpp

```diff
@@ -939,15 +1122,25 @@ calcMuscleDynamicsInfo(const SimTK::State& s, MuscleDynamicsInfo& mdi) const
         if(fiberStateClamped < 0.5) { //flag is set to 0.0 or 1.0
             SimTK::Vec4 fiberForceV;
 
-            fiberForceV = calcFiberForce(fiso, a,
-                                         mli.fiberActiveForceLengthMultiplier,
-                                         mvi.fiberForceVelocityMultiplier,
-                                         mli.fiberPassiveForceLengthMultiplier,
-                                         mvi.normFiberVelocity);
-            fm   = fiberForceV[0];
-            aFm  = fiberForceV[1];
-            p1Fm = fiberForceV[2];
-            p2Fm = fiberForceV[3];
+            aFm = CalcActiveFiberForce(
+                fiso,
+                a,
+                mli.fiberActiveForceLengthMultiplier,
+                mvi.fiberForceVelocityMultiplier);
+            p1Fm = CalcPassiveFiberElasticForce(
+                fiso,
+                mli.fiberPassiveForceLengthMultiplier);
+            p2Fm =
+                CalcPassiveFiberDampingForce(fiso, mvi.normFiberVelocity, beta);
+            fm = CalcFiberForce(
```

Comment:
I would expect the compiler to optimize, and remove the repeated calls, I even hinted with `inline`! Should i verify? Or just not do this?

Reply:
resolved it differently, since it was just `fm = aFm + pFm`,

I split it into different calls, because somewhere the total was not used, and in most places the parts were not used. So I think currently it has no repeated computations, and is not calculating more than needed.